File: credit-engine-api/src/main/java/com/credit/engine/application/service/receivable/ReceivableService.java
package com.credit.engine.application.service.receivable;

import com.credit.engine.application.dto.receivable.ReceivableRequest;
import com.credit.engine.application.dto.receivable.ReceivableResponse;

import java.util.List;
import java.util.UUID;

public interface ReceivableService {

    ReceivableResponse create(ReceivableRequest request);

    ReceivableResponse update(UUID id, ReceivableRequest request);

    ReceivableResponse findById(UUID id);

    List<ReceivableResponse> findByAssignor(UUID assignorId);

    List<ReceivableResponse> findAll();

}

File: credit-engine-ui/src/services/receivable/receivable.service.ts
import { http } from '../../api/client';
import DateUtils from '../../utils/DateUtils';
import type ReceivableRequest from './dto/ReceivableRequest';
import type ReceivableResponse from './dto/ReceivableResponse';

const controllerUrl = '/receivables';

// dueDate e um LocalDate no backend. Convertemos para yyyy-MM-dd aqui para
// nao serializar um Instant completo via JSON.stringify(Date).
function toPayload(request: ReceivableRequest) {
    return {
        ...request,
        currencyCode: request.currencyCode.toUpperCase(),
        dueDate: request.dueDate ? DateUtils.toLocalDateString(request.dueDate) : null,
    };
}

export const receivableService = {

    findByAssignor(assignorId: string) {
        return http.get<ReceivableResponse[]>(`${controllerUrl}?assignorId=${assignorId}`);
    },

    create(request: ReceivableRequest) {
        return http.post<ReceivableResponse>(controllerUrl, toPayload(request));
    },

    update(id: string, request: ReceivableRequest) {
        return http.put<ReceivableResponse>(`${controllerUrl}/${id}`, toPayload(request));
    },

};
